gtsam/discrete/DiscreteConditional.cpp

```diff
@@ -478,6 +478,19 @@ double DiscreteConditional::evaluate(const HybridValues& x) const {
   return this->evaluate(x.discrete());
 }
 
+/* ************************************************************************* */
+DiscreteFactor::shared_ptr DiscreteConditional::max(
+    const Ordering& keys) const {
+  return BaseFactor::max(keys);
```

Comment:
If this just returns BaseFactor::max, why does this method exist?

Reply:
So we can overload it within `TableDistribution`. The original signature for `max` is inherited from DecisionTreeFactor as
```cpp
DecisionTreeFactor::shared_ptr max(const Ordering& keys) const;
```
which returns a DecisionTreeFactor. It won't work if we need to return a TableFactor, and changing the return type in DecisionTreeFactor is not backwards compatible (needs an explicit cast).